src/audiovideocontroller/DefaultAudioVideoController.ts

```diff
@@ -286,6 +287,7 @@ export default class DefaultAudioVideoController implements AudioVideoController
               new FinishGatheringICECandidatesTask(this.meetingSessionContext),
               new SubscribeAndReceiveSubscribeAckTask(this.meetingSessionContext),
               new SetRemoteDescriptionTask(this.meetingSessionContext),
+              new ValidateAttendeePresenceTask(this.meetingSessionContext),
```

Comment:
Should we make the time out a configurable value in `MeetingSessionConfiguration` ?
I would like this feature to have a less blast radius, e.g the default `ValidateAttendeePresenceTask` is a no-op, if user configure the time out then this task will then wait for attendee presence

Reply:
That's a good suggestion. I can make it configurable. Why do you want to have a smaller blast radius?
- `null` (default) : no-op. SDK will not restart the session.
- `number`: SDK will wait the given amount of time and restart if not present.

One more question - do you see any issue programmatically choosing audio device when SDK restarts for the no-presence event?

```
const devices = await app.audioVideo.listAudioInputDevices();
if (!devices.length) {
  await app.audioVideo.chooseAudioInputDevice(devices[0].deviceId);  
}
```

Reply:
> Why do you want to have a smaller blast radius?
null (default) : no-op. SDK will not restart the session.
number: SDK will wait the given amount of time and restart if not present.

I think I am being super cautious here about developers having an option to disable this safety check. But on second thought, "attendee presence" is pretty vital and we should validate this. So it's up to you :)


>One more question - do you see any issue programmatically choosing audio device when SDK restarts for the no-presence event?
const devices = await app.audioVideo.listAudioInputDevices();
if (!devices.length) {
  await app.audioVideo.chooseAudioInputDevice(devices[0].deviceId);  
}

Where is the code ? Is it demo application ?

Reply:
We don't have that code yet. If you don't call `chooseAudioInputDevice`, SDK joins a meeting with an empty audio device.

In Safari, if you join a meeting with an empty audio device, an attendee is not present. To mitigate this issue, SDK can explicitly call `chooseAudioInputDevice` before restarting the session between line 302 and 303 here.

https://github.com/aws/amazon-chime-sdk-js/blob/master/src/audiovideocontroller/DefaultAudioVideoController.ts#L302-L303

Reply:
might need some input from @aburnson 

My first impression is that `chooseAudioInputDevice` on a real device should only be triggered by user action, if SDK calls `chooseAudioInputDevice` in this case

Reply:
Device selection should only occur by a user via the developer's application. The SDK should not assume a device since that could lead to leaking data from an unintended device. I think it could be reasonable to call `chooseAudioInputDevice` for the previously selected device (or a null device) if the intention is to try to kick the session join through again.

Reply:
Thanks for the suggestions. In this PR, I'll not call `chooseAudioInputDevice` during reconnect. I'll push a new revision with a configurable timeout.